en/tutorials-and-examples/blog/part-three.rst

```diff
@@ -0,0 +1,267 @@
+
+            $categories = $this->table('categories');
+            $categories->addColumn('parent_id', 'integer', ['null' => true, 'default' => null])
+                ->addColumn('lft', 'integer', ['null' => true, 'default' => null])
+                ->addColumn('rght', 'integer', ['null' => true, 'default' => null])
+                ->addColumn('name', 'string', ['limit' => 255])
+                ->addColumn('description', 'string', ['limit' => 255, 'null' => true, 'default' => null])
+                ->addColumn('created', 'datetime')
+                ->addColumn('modified', 'datetime', ['null' => true, 'default' => null])
+                ->save();
+        }
+    }
+
+Now run the following command to create your tables::
+
+    bin/cake migrations migrate
+
+
+Modifying the Tables
+====================
+
+With our tables set up, we can now focus on categorizing our articles.
+
+We suppose you already have the files (Tables, Controllers and Templates of
+Articles) from part 2. So we'll just add the references to categories.
+
+We need to associated the Articles and Categories tables together. Open
+the ``/src/Model/Table/ArticlesTable.php`` file and add the following::
+
+    // src/Model/Table/ArticlesTable.php
+
+    namespace App\Model\Table;
+
+    use Cake\ORM\Table;
+
+    class ArticlesTable extends Table
+    {
+        public function initialize(array $config)
+        {
+            $this->addBehavior('Timestamp');
+            // Just add the belongsTo relation with CategoriesTable
+            $this->belongsTo('Categories', [
+                'foreignKey' => 'category_id',
+            ]);
+        }
+    }
+
+Generate Skeleton Code for Categories
+=====================================
+
+Create all files by launching bake commands::
+
+    bin/cake bake model Categories
+    bin/cake bake controller Categories
+    bin/cake bake view Categories
+
+The bake tool has created all your files in a snap. You can give them a quick
+read if you want re-familiarize yourself with how CakePHP works.
+
+Attach TreeBehavior to CategoriesTable
+======================================
+
+The :doc:`TreeBehavior </orm/behaviors/tree>` helps you manage hierarchical Tree
+structures in database table. It uses the `MPTT logic
+<http://www.sitepoint.com/hierarchical-data-database-2/>`_ to manage the data.
+MPTT tree structures are optimized for reads, which often makes them a good fit
+for read heavy applications like blogs.
+
+If you open the ``src/Model/Table/CategoriesTable.php`` file, you'll see
+that the TreeBehavior has been attached to your CategoriesTable in the
+``initialize`` method::
+
+    $this->addBehavior('Tree');
+
+With the TreeBehavior attached you'll be able to access some features like
+reordering the categories.  We'll see that in a moment.
+
+But for now, you have to remove the following inputs in your Categories add and
+edit template files::
+
+    echo $this->Form->input('lft');
+    echo $this->Form->input('rght');
```

Comment:
Leaving them out makes sense to me. Generally I don't think we want people manually managing those fields.